### PR TITLE
Update packages, add bundle analyzer, add declarationMap

### PR DESCRIPTION
**Pull request by similiarity**
----

> [!IMPORTANT]
> Update dependencies, add bundle analyzer, and enhance `ProfileImageEditor` with `react-easy-crop`.
> 
>   - **Dependencies**:
>     - Update `lucide-react` to `^0.508.0` in `apps/dashboard/package.json`, `packages/stack-emails/package.json`, and `packages/stack-ui/package.json`.
>     - Replace `react-avatar-editor` with `react-easy-crop` in `packages/react/package.json`, `packages/stack/package.json`, and `packages/template/package-template.json`.
>     - Add `@next/bundle-analyzer` to `examples/demo/package.json`.
>   - **Build Configuration**:
>     - Add bundle analyzer configuration to `examples/demo/next.config.js`.
>     - Add `analyze` script to `examples/demo/package.json`.
>   - **Component Changes**:
>     - Replace `AvatarEditor` with `Cropper` in `ProfileImageEditor` in `packages/template/src/components/profile-image-editor.tsx`.
>     - Update `Calendar` component styles in `packages/stack-ui/src/components/ui/calendar.tsx`.
>   - **Utility Functions**:
>     - Update `isBase32`, `isBase64`, and `isBase64Url` test cases in `bytes.tsx` to handle specific edge cases.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=stack-auth%2Fstack-auth&utm_source=github&utm_medium=referral)<sup> for ae4574dba0a35785910e84e0e37679edd7f6fc96. You can [customize](https://app.ellipsis.dev/stack-auth/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: packages/template/src/components/profile-image-editor.tsx [hunk 1]
```diff
@@ -74,40 +133,46 @@ export function ProfileImageEditor(props: {
...
-        defaultValue={[slideValue]}
-        value={[slideValue]}
-        onValueChange={(v) => setSlideValue(v[0])}
+        value={[zoom]}
+        onValueChange={(v) => onZoomChange(v[0])}
       />
 
       <div className='flex flex-row gap-2'>
         <Button
           onClick={async () => {
```
[the remark this answers is not in the document]
Comment (similiarity) [reply packages/template/src/components/profile-image-editor.tsx:159]: The original also didn't have it, should I add it?

**Review by fomalhautb — APPROVED** [review]

**Review by N2D4 — APPROVED** [review]
